nntrainer/graph/network_graph.cpp

```diff
@@ -494,6 +511,47 @@ void NetworkGraph::backwarding(
     backwarding_op(ln, iteration);
     END_PROFILE(profile_keys.at(ln->getType()));
   }
+
+  /** perform clipping of the gradients by global norm if any */
+  auto clip_weights = tensor_manager->getWeights([](Weight *w) {
+    return w->hasGradient() && w->isGradientLastAccess() &&
+           w->isGradientClipByGlobalNorm();
+  });
+  /** calculate the global norm */
+  unsigned int global_norm = 0;
+  for (auto w : clip_weights) {
+    global_norm += std::pow(w->getGradientNorm(), 2);
```

Comment:
I recommend to make a tensor to store the vector to calculate global_norm
and global_norm = norms.l2norm()

There are two major factors behind.

1. Accumulation can go out of float range easily, so this might be unstable.
2. Speed (it was observed that this is significantly slower then Tensor::l2norm from the eariler version of nntrainer)

Reply:
fair enough, I will update.